app/api-v1/routes/attachment.js

```diff
@@ -8,17 +9,17 @@ module.exports = function (attachmentService) {
     POST: async function (req, res) {
       logger.info('Attachment upload: ', req.file)
 
-      if (!req.file) {
-        res.status(400).json({ message: 'No file uploaded' })
-      }
-
       try {
+        if (!req.file) {
+          throw new FileUploadError({ code: 400, message: 'No file uploaded', service: 'attachment' })
+        }
+
         const attachment = await attachmentService.createAttachment(req.file)
         const result = attachment[0]
         res.status(201).json({ ...result, size: req.file.size })
       } catch (err) {
         logger.warn('Error in POST /attachment: %s', err.message)
-        res.status(500).send(err.message)
+        res.status(err.code).send(err.message)
```

Comment:
you may want to be cautious about assuming code exists. Maybe default to 500 if it isn't a number?

Reply:
added `|| 500`

Reply:
do we not need to consider some of switch patten here?

Reply:
this should really be defined somewhere already

Reply:
one option would be to add a wrapper function to the `errors` util that we wrap around each handler. That can then handle the switch across the different error types?

Reply:
something like
```
switch(err.type){
    case UploadFileError:
        res.status(err.code).send(err.message)
    default:
       res.status(500).send(err.message)
}
```

Reply:
just commited something like ^